ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/CancelFunctionButton/CancelFunctionModal.tsx

```diff
@@ -0,0 +1,181 @@
+import { useCallback, useState } from 'react';
+import { Alert } from '@inngest/components/Alert';
+import { NewButton } from '@inngest/components/Button';
+import { RangePicker } from '@inngest/components/DatePicker';
+import { Input } from '@inngest/components/Forms/Input';
+import { Modal } from '@inngest/components/Modal';
+import { subtractDuration } from '@inngest/components/utils/date';
+
+import { usePlanFeatures } from '../runs/usePlanFeatures';
+import { useCreateCancellation } from './useCreateCancellation';
+import { useRunCount, type RunCountInput } from './useRunCount';
+
+type Props = {
+  envID: string;
+  functionSlug: string;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export function CancelFunctionModal(props: Props) {
+  const { envID, functionSlug, isOpen, onClose: _onClose } = props;
+  const [creationError, setCreationError] = useState<Error>();
+  const [isCreating, setIsCreating] = useState<boolean>(false);
+  const [name, setName] = useState<string>();
+  const [timeRange, setTimeRange] = useState<{ start: Date; end: Date }>();
+  const createCancellation = useCreateCancellation({ functionSlug });
+
+  const onClose = useCallback(() => {
+    setCreationError(undefined);
+    setIsCreating(false);
+    setName(undefined);
+    setTimeRange(undefined);
+
+    _onClose();
+  }, [_onClose]);
+
+  const onSubmit = useCallback(async () => {
+    if (!timeRange) {
+      return;
+    }
+
+    try {
+      const res = await createCancellation({
+        name,
+        queuedAtMax: timeRange.end,
+        queuedAtMin: timeRange.start,
+      });
+      if (res.error) {
+        throw res.error;
+      }
+      onClose();
+    } catch (e) {
+      if (e instanceof Error) {
+        setCreationError(e);
+      } else {
+        setCreationError(new Error('unknown error'));
+      }
+      return;
+    } finally {
+      setIsCreating(false);
+    }
+  }, [createCancellation, name, onClose, timeRange]);
+
+  let runCountInput: RunCountInput | undefined;
+  if (timeRange) {
+    runCountInput = {
+      envID,
+      functionSlug,
+      queuedAtMin: timeRange.start,
+      queuedAtMax: timeRange.end,
+    };
+  }
+
+  const countRes = useRunCount(runCountInput);
+  const featureRes = usePlanFeatures();
+
+  let upgradeCutoff: Date | undefined;
```

Comment:
Just a style/readability thing, but the `let` + `if` setup implies to me that this var might also be set somewhere else downstream...whereas a `const` + `ternary` tells me the thing is resolved here and done. Though I know it's more terse. Like: 

```
const upgradeCutoff: Date | undefined = featureRes.data
    ? subtractDuration(new Date(), { days: featureRes.data.history })
    : undefined;
```

Reply:
I think ternaries have poor readability so I like to avoid them